src/curl_https_download_file.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+	Copyright (C) 2003 - 2022
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#include "curl_https_download_file.hpp"
+
+#include "filesystem.hpp"
+#include "log.hpp"
+
+#include <curl/curl.h>
+
+static lg::log_domain log_network("network");
+#define ERR_NW LOG_STREAM(err, log_network)
+#define DBG_NW LOG_STREAM(debug, log_network)
+
+namespace network
+{
+    static size_t write_callback(char* contents, size_t size, size_t nmemb, void* buffer)
```

Comment:
Does this need  to be `static`?

Reply:
Looks like it doesn't.

Reply:
Actually, apparently macOS requires it to be static.